libtenzir/include/tenzir/detail/weak_run_delayed.hpp

```diff
@@ -35,15 +35,20 @@ auto weak_run_delayed(caf::scheduled_actor* self, caf::timespan delay,
 /// @param self The hosting actor pointer.
 /// @param delay The delay after which to repeat the action.
 /// @param function The action to run with the signature () -> void.
+/// @param run_immediately Whether to run the function immediately.
 template <class Function>
   requires std::is_invocable_r_v<void, std::remove_reference_t<Function>&>
 void weak_run_delayed_loop(caf::scheduled_actor* self, caf::timespan delay,
-                           Function&& function) {
+                           Function&& function, bool run_immediately = true) {
+  if (run_immediately) {
+    std::invoke(function);
+  }
```

Comment:
I checked all uses of this function whether they can handle being run immediately, as this might be dangerous during the setup. But it _seems_ like they do.

Not sure whether this is a good default. In most cases it's unnecessary. For collecting health metrics it makes sense of course. 

(But I don't care too much, please just resolve if you want to keep this as-is.)

Reply:
I also checked all the cases when I made this change, and I found that almost all of them actually want to be run immediately. I considered creating a second function instead, but then we wouldn't use the old one anymore.